solenoid_app/solenoid_math/solver.py

```diff
@@ -245,49 +306,161 @@ def solenoid_performance(volts, length, r0, ra, gauge, location, force):
+            value = i * ureg(idv_unit)
+            try:
+                result.append((i, solenoid_convert(volts, length, r0, ra, gauge, location, value, output_unit)))
+            except NoSolution:
+                pass
 
-    elif idv == "gauge":
+    elif idv == "awg":
         for item in AWG_DATA.keys():
-            result.append((item, solenoid_performance(volts, length, r0, ra, item, location, force)))
+            try:
+                result.append((item, solenoid_convert(volts, length, r0, ra, item, location, force, output_unit)))
+            except NoSolution:
+                pass
 
     elif idv == "x":
         for i in list(np.arange(start, stop, step)):
-            result.append((i, solenoid_performance(volts, length, r0, ra, gauge, i, force)))
+            value = i * ureg(idv_unit)
+            try:
+                result.append((i, solenoid_convert(volts, length, r0, ra, gauge, value, force, output_unit)))
+            except NoSolution:
+                pass
 
     return result
+
+
+"""
+Wrapper for solenoid_performance. This function takes in values of differing units, converts them to base SI units, then
+solves for the specified parameter. When a solution is found, it is converted back to the expected output units.
+
+The variable being solved for is inputted as a None value. All other
+arguments are then required and cannot be None
+
+Parameters:
+    volts (float | None): Voltage applied to the solenoid - Volts
+    length (pint Quantity | None): Overall length of the solenoid coil
+    r0 (pint Quantity | None): Inner radius of the solenoid coil
+    ra (pint Quantity | None): Outer radius of the solenoid coil
+    gauge (string): A value between "0000" -> "40"
+    location (pint Quantity | None): Location (Stroke) of the solenoid core within the coil
+    force (pint Quantity | None): The force produced by the solenoid
+    output_unit (string): unit identifier string the result - e.x.: "mm", "meters", "lbf", etc...
+
+Returns:
+    result (float): The solved value of specified variable
+"""
+def solenoid_convert(volts, length, r0, ra, gauge, location, force, output_unit):
+
+    chosen = None
+
+    # convert all entered parameters to base SI units
+    if length is not None:
+        length = length.to_base_units()
+        length = float(length.magnitude)
+    else:
+        chosen = "length"
+
+    if r0 is not None:
+        r0 = r0.to_base_units()
+        r0 = float(r0.magnitude)
+    else:
+        chosen = "r0"
+
+    if ra is not None:
+        ra = ra.to_base_units()
+        ra = float(ra.magnitude)
+    else:
+        chosen = "ra"
+
+    if location is not None:
+        location = location.to_base_units()
+        location = float(location.magnitude)
+    else:
+        chosen = "location"
+
+    if force is not None:
+        force = force.to_base_units()
+        force = float(force.magnitude)
+    else:
+        chosen = "force"
```

Comment:
This seems repetitive, however I can't think of a real good way to fix it. It would require iterating over those parameters or calling a separate helper function. If you can think of a clean way, do it! If not, everything else LGTM

Reply:
One way that I had on my local was to convert all the variables inside of the solenoid_range function before we solve for range.  The solenoid_convert only function is to just perform conversion without having to call on solenoid_performance.  That  way the conversion for the majority of the variable happen only once.  I ran a simple performance test with the time library and the difference weren't that much so we stick with this since we know it's working.

Reply:
+1 to this. There could be a functional way to do this I believe. 

Ideally we could take out volts, find which is None from the start, and have a list of everything else which we could do a map on -> 
```python3
def convert(arg):
    arg = arg.to_base_units()
    arg = float(arg.magnitude)
    return arg

parameters = map(lambda x: x=convert(x), parameters)
```
Something like the above I think is a bit cleaner, but if it's too much overhead I would lean towards the side of readability

Reply:
So something like throw all of the converted numbers into a tuple and return it so have a copy for instant access?

Reply:
Yeah that sounds right to me. Lmk if you want to pair program it

Reply:
Might want to have it be a dictionary instead

Reply:
@trevorliberty Just made a change to make it simpler. Let me know what you think.

Reply:
Looks good